components/fonts/platform/freetype/font.rs

```diff
@@ -330,22 +333,22 @@ impl PlatformFontMethods for PlatformFont {
         // 0.5em should be used."
         let mut x_height = 0.5 * em_height;
         let mut average_advance = 0.0;
-        if let Some(os2) = face_ptr.os2_table() {
-            if !os2.y_strikeout_size.is_zero() && !os2.y_strikeout_position.is_zero() {
-                strikeout_size = os2.y_strikeout_size as f64 * y_scale;
-                strikeout_offset = os2.y_strikeout_position as f64 * y_scale;
+
+        if let Ok(os2) = font_ref.and_then(|font_ref| font_ref.os2()) {
+            let y_strikeout_size = os2.y_strikeout_size();
+            let y_strikeout_position = os2.y_strikeout_position();
+            if !y_strikeout_size.is_zero() && !y_strikeout_position.is_zero() {
+                strikeout_size = y_strikeout_size as f64 * y_scale;
+                strikeout_offset = y_strikeout_position as f64 * y_scale;
             }
-            if !os2.sx_height.is_zero() {
-                x_height = os2.sx_height as f64 * y_scale;
+
+            if let Some(sx_height) = os2.sx_height() {
+                x_height = sx_height as f64 * y_scale;
```

Comment:
The previous code was guarded by `!os2.sx_height.is_zero()`, but here we don't ensure sx_height is non-zero. Are both versions equivalent?

Reply:
I'm not sure what kind of fonts have a value of zero here, but I've restored the zero check here, just in case it matters. I think only Firefox was doing this check in the past and we inherited it from them.